### DISCUSSION TITLE
Any plans to bundle this in Node?

### DISCUSSION THREAD

**Discussion by alvinometric** [nodejs/undici#1512] [General]
Hello,

It's fantastic to see an official HTTP library from the Node team. Are there any plans to bundle this with Node itself? Just like `fs`, for example.

Right now you have to install a package so it's hard to differentiate `undici` compared to the other libraries out there.

Thank you

**Comment by mcollina** [comment]
We plan to but we have no timeline.

**Comment by alvinometric** [comment]
Great, thank you for the quick reply

**Comment by cristobal** [comment]
Would be nice to see this land in Node 20 in October when it takes over as LTS 🙏🏽

**Comment by tosinamuda** [comment]
is this bundled in Node 21 already?

**Comment by cristobal** [comment]
So @mcollina any news/timeline on bundling undici natively, which is now clearly pretty stable in nodejs. Would be nice not to have to install this as an additional package.

Any sane production project that uses fetch server side need to tune:
* connection pooling with keep-alive and
* concurrent connections

to avoid high usage of sockets due to burst of traffic, the defaults don't scale, and to tune these settings one need to install undici as a separate library.
